### DISCUSSION TITLE
Terraform provider backend configuration

### DISCUSSION THREAD

**Discussion by andreaspoldi** [nitrictech/nitric#711] [Q&A] [answered]
Hi, 
I was wondering if there is a way to override the "local" backend option for terraform state when using Terraform as deployment provider in Nitric. From the code it seems that there is at least an option for S3 as provider, but I cannot understand how to work it out...

Thanks

**Comment by jyecusch** [comment]
For the Terraform provider Nitric generates a fairly standard Terraform project, so you can manage the state just as you normally would with Terraform

**Reply by andreaspoldi** [reply]
thanks for the reply...yes I am aware of that, what I mean is that with "nitric up" I get a json tf template with the state file created (or somehow hardcoded) as standard local state file 

`  "terraform": {
    "backend": {
      "local": {
        "path": "/Users/.../nitric/websocket-app/terraform.websocket-app-dev.tfstate"
      }
    },`

I'd have to edit my generated terraform template in order to make it work with a S3 backend (or anything else than a local one)

**Reply by jyecusch** [reply]
Sorry, I misread the question. Tim has provided a more detailed answer, if there is other config you think should be added, let us know.

**Comment by tjholm** [comment, marked as answer]
@andreaspoldi this has been released in `v1.16.10`. Current supported backends are `s3` and `gcs`.

Configuration can be made in the nitric stack file and looks something like:

As an example an s3 backend can be configured like so:
```yaml
backend:
  s3:
    bucket: "mybucket"
    key: "path/to/my/key"
    region: "us-east-1"
```
This config is passed through directly the cdktf so any options valid for https://developer.hashicorp.com/terraform/language/backend/s3 should work.

Let me know if you have any issues with that config.

**Reply by andreaspoldi** [reply]
Ok that's look great, thank you
